pkgs/top-level/all-packages.nix

```diff
@@ -12440,17 +12444,73 @@ with pkgs;
 
   openmpi = callPackage ../development/libraries/openmpi { };
 
-  ouch = callPackage ../tools/compression/ouch { };
+  openmpi-cuda = openmpi.override {
+    cudaSupport = true;
+    rocmSupport = false;
+    ucx = ucx-cuda;
+    ucc = ucc-cuda;
+  };
 
-  outils = callPackage ../tools/misc/outils { };
+  # We need to use the 5.0.x branch until it is released
+  openmpi-rocm = (openmpi.overrideAttrs (final: prev: {
```

Comment:
This is a very brittle construction, which most likely breaks when openmpi is updated next (either one, 4 or 5). Having such a lengthy override in all-packages make this big file even worse. 
You could make a separate file for openmpi-5. However, I would rather prefer to wait until openmpi-5 is released (it hopefully does not take that long anymore). That is usually how I personally handle it for my packages and developments.

Is there any reason that speaks against having openmpi-5 with rocm support out-of-tree until it is stable?

Reply:
A separate file like pmix may be for the best. I kind of threw these overrides together.